cmd/util/cmd/execution-state-extract/cmd.go

```diff
@@ -148,14 +209,65 @@ func run(*cobra.Command, []string) {
 		log.Warn().Msgf("atree migration has verbose validation error logging enabled which may increase size of log")
 	}
 
-	err := extractExecutionState(
-		log.Logger,
-		flagExecutionStateDir,
-		stateCommitment,
-		flagOutputDir,
-		flagNWorker,
-		!flagNoMigration,
-	)
+	var inputMsg string
+	if len(flagInputPayloadFileName) > 0 {
+		// Input is payloads
+		inputMsg = fmt.Sprintf("reading payloads from %s", flagInputPayloadFileName)
+	} else {
+		// Input is execution state
+		inputMsg = fmt.Sprintf("reading block state commitment %s from %s",
+			hex.EncodeToString(stateCommitment[:]),
+			flagExecutionStateDir,
+		)
+	}
+
+	var outputMsg string
+	if len(flagOutputPayloadFileName) > 0 {
+		// Output is payload file
+		if len(exportedAddresses) == 0 {
+			outputMsg = fmt.Sprintf("exporting all payloads to %s", flagOutputPayloadFileName)
+		} else {
+			outputMsg = fmt.Sprintf(
```

Comment:
now an output payload file can possibly include only a subset of all the addresses, which means a payload file can possibly include a subset of world state. This is useful for debugging, but also introduce a chance of a wrong payload file used for state extraction, which only include subset of the world state. 

Yes, we have include the payload count encoded in the payload file, but it's only used for validating the file integrity instead of ensuring the payloads include the full set of the world state. 

Should we add something to the payload file to ensure that a payload file generated by a subset of world state will never be used for state extraction? 

Thoughts?

Reply:
> This is useful for debugging, but also introduce a chance of a wrong payload file used for state extraction, which only include subset of the world state.

Great point!  :+1: 

I'll modify payload file format to include a flag indicating whether payloads represent partial/full state.  I'll also add a flag in state extraction program to explicitly allow partial state as input so we reduce chance of user error.

These changes will be in a separate PR combined with hashed payload file mentioned earlier.